### PR TITLE
Icerockdev#19 mapbox feature

### PR DESCRIPTION
**Pull request by Limonlol**

### CODE REVIEW

File: maps-google/build.gradle.kts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Alex009) [inline maps-google/build.gradle.kts]: invalid change :)

File: maps-mapbox/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/build.gradle.kts]: now 2020 :)

File: maps-mapbox/src/androidMain/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest package="dev.icerock.moko.maps.google" />
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/AndroidManifest.xml]: maps.mapbox

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt]: copyright here please

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+import dev.icerock.moko.geo.LatLng
+import dev.icerock.moko.graphics.Color
+import dev.icerock.moko.maps.*
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt]: disable star import please

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt]: copyright here please

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt]: copyright here please

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt [hunk 8]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: 2020 year :)

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package dev.icerock.moko.maps.mapbox
+
+import dev.icerock.moko.geo.LatLng
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: empty file, remove please

File: maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt [hunk 10]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt]: 2020

File: maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt [hunk 11]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt]: 2020

File: maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt [hunk 12]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt]: 2020

File: maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/UiSettings.kt [hunk 13]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/commonMain/kotlin/dev/icerock/moko/maps/mapbox/UiSettings.kt]: 2020

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/Async.kt [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+import kotlinx.cinterop.staticCFunction
+import platform.Foundation.NSThread
+import platform.darwin.dispatch_get_main_queue
+import platform.darwin.dispatch_sync_f
+import kotlin.native.concurrent.Continuation2
+import kotlin.native.concurrent.callContinuation2
+
+@Suppress("NOTHING_TO_INLINE")
+internal inline fun <T1, T2> mainContinuation(
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/Async.kt]: it's used?

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/GDirection.kt [hunk 15]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/GDirection.kt]: 2020

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/GDirection.kt [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/GDirection.kt]: but all file content is only for google maps parsing. remove file

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxAnnotation.kt [hunk 17]
```diff
@@ -0,0 +1,9 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxAnnotation.kt]: copyright

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt [hunk 18]
```diff
@@ -0,0 +1,190 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt]: copyright

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+
+import cocoapods.Mapbox.*
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxController.kt]: disable star import

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt [hunk 20]
```diff
@@ -0,0 +1,41 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxMarker.kt]: copyright

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt [hunk 21]
```diff
@@ -0,0 +1,9 @@
+package dev.icerock.moko.maps.mapbox
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt]: copyright

File: maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt [hunk 22]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/src/iosMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: 2020

File: sample/ios-app/Podfile [hunk 23]
```diff
@@ -12,4 +12,6 @@ platform :ios, '11.0'
 target 'TestProj' do
   # MultiPlatformLibrary
   pod 'MultiPlatformLibrary', :path => '../mpp-library'
+  pod 'Mapbox-iOS-SDK'
```
Comment (Alex009) [inline sample/ios-app/Podfile]: set version. 5.5.0 now used

File: sample/ios-app/src/AppDelegate.swift [hunk 24]
```diff
@@ -12,10 +12,6 @@ class AppDelegate: NSObject, UIApplicationDelegate {
     var window: UIWindow?
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
-        
-        // TODO: Replace with your API Key from https://developers.google.com/maps/documentation/ios-sdk/
-        GMSServices.provideAPIKey("YOUR-API-KEY")
```
Comment (Alex009) [inline sample/ios-app/src/AppDelegate.swift]: not remove it :) just comment locally for testing mapbox only, but not remove

File: sample/ios-app/src/TestViewController.swift [hunk 25]
```diff
@@ -4,36 +4,26 @@
 
 import UIKit
 import MultiPlatformLibrary
-import GoogleMaps
+import Mapbox
 
-class TestViewController: UIViewController {
+class TestViewController: UIViewController, MGLMapViewDelegate {
```
Comment (Alex009) [inline sample/ios-app/src/TestViewController.swift]: please not broke test sample, just add new viewController with mapbox, google should not be removed.

File: sample/mpp-library/build.gradle.kts [hunk 26]
```diff
@@ -48,9 +49,9 @@ kotlin {
         .forEach { framework ->
             framework.isStatic = true
 
-            val frameworks = listOf("Base", "Maps").map { frameworkPath ->
-                project.file("../ios-app/Pods/GoogleMaps/$frameworkPath/Frameworks").path.let { "-F$it" }
-            }
```
Comment (Alex009) [inline sample/mpp-library/build.gradle.kts]: add, not replace

File: sample/mpp-library/build.gradle.kts [hunk 27]
```diff
@@ -19,7 +19,7 @@ val libs = listOf(
     Deps.Libs.MultiPlatform.mokoMvvm,
     Deps.Libs.MultiPlatform.mokoPermissions,
     Deps.Libs.MultiPlatform.mokoMaps,
-    Deps.Libs.MultiPlatform.mokoMapsGoogle
```
Comment (Alex009) [inline sample/mpp-library/build.gradle.kts]: add, not replace

File: sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/TrackerViewModel.kt [hunk 28]
```diff
@@ -6,23 +6,26 @@ package com.icerockdev.library
...
-import dev.icerock.moko.graphics.Color
 import dev.icerock.moko.maps.ZoomConfig
-import dev.icerock.moko.maps.google.GoogleMapController
-import dev.icerock.moko.maps.google.UiSettings
+import dev.icerock.moko.maps.mapbox.MapboxController
+import dev.icerock.moko.maps.mapbox.UiSettings
 import dev.icerock.moko.mvvm.viewmodel.ViewModel
 import kotlinx.coroutines.launch
 
 class TrackerViewModel(
```
Comment (Alex009) [inline sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/TrackerViewModel.kt]: add other sample viewmodel, not broke google testing

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt (hunk 9 above)
Comment (Limonlol) [reply maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: this is android utils. i just comment it, cause there was error in build.

File: sample/ios-app/src/GoogleMapViewController.swift [hunk 29]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2010 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline sample/ios-app/src/GoogleMapViewController.swift]: 2020)

File: sample/ios-app/src/Info.plist [hunk 30]
```diff
@@ -60,5 +60,7 @@
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationPortraitUpsideDown</string>
 	</array>
+	<key>MGLMapboxAccessToken</key>
+	<string>pk.eyJ1IjoibHV2IiwiYSI6ImNpZmkyNzQzbDAwdmN1Nmx5cnRkZno5NDkifQ.nR8a_kXuMZEHjn404_A_kQ</string>
```
Comment (Alex009) [inline sample/ios-app/src/Info.plist]: it's testing key or production?

File: sample/ios-app/src/MapboxViewController.swift [hunk 31]
```diff
@@ -0,0 +1,34 @@
+/*
+* Copyright 2010 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline sample/ios-app/src/MapboxViewController.swift]: 2020

**Pull request merged**